datageneration/relation_generator.py

```diff
@@ -59,6 +60,37 @@ def generate_in_area(self, num_entities: int) -> None:
         It returns None, that indicates that the relation is not clear or one object exists
         '''
         return None
+
+    def get_task(self, num_entities: int):
+        """
+        This method of selecting the task of the query using an exponential decay method. It first filters out
+        tasks that are not viable due to the number of entities. It then selects the task based on a probability
+        distribution that assigns higher probabilities to task at the beginning of the list. This allows for
+        a drafting system that prioritises certain tasks over others, as e.g. "individual_distances" is a
+        far more difficult task than "in_area" and therefore requires more training data.
+
+        Example probability distribution with decay of 0.5:
+            - 3 or more entities: [0.50648039 0.30719589 0.18632372]
+            - 2 entities: [0.62245933 0.37754067]
+            - 1 entity: [1.0]
+
+        :param num_entities: The number of entities of the query
+        :return: The selected task
+        """
+        viable_tasks = [RELATION_TASKS.INDIVIDUAL_DISTANCES.value, RELATION_TASKS.WITHIN_RADIUS.value,
```

Comment:
we have a list doing this as self.tasks, maybe you can check this. You can use sth like self.tasks[:-1] as you do for viable tasks. decay_rate should be moved to under def __init__ since it is a hard-coding value.

Reply:
This does work, it is however important to keep the order exactly as it is. It might become a problem if the order is switch around at another part of the code, because the impact on this method is not obvious. I would therefore stick to this variant, as future coders will immediately know to not change the order.